src/Symfony/Component/HttpKernel/Controller/ErrorController.php

```diff
@@ -38,6 +39,13 @@ public function __construct(HttpKernelInterface $kernel, string|object|array|nul
 
     public function __invoke(\Throwable $exception): Response
     {
+        foreach (ErrorHandler::getErrorEnhancers() as $errorEnhancer) {
+            if ($e = $errorEnhancer->enhance($exception)) {
+                $exception = $e;
+                break;
+            }
+        }
+
```

Comment:
Before this PR, the error enhancement will depend on the `framework.handle_all_throwables` config, if `true` the `Error` instance will be rendered through the renderers system directly (skipping the error enhancement), otherwise the `ErrorHandler::handleException()` will address it (the error enhancement happens here) and then it will go through the renderers system. That's why it's happening. If you change the config to `false` you will see the enhanced error.

Now I'm wondering if it's better to add a new static method inside the ErrorHandler to do this job, thus avoiding duplicated logic (I don't remember the reason but it's currently skipping instances of `OutOfMemoryError`).

Maybe something like this:
```php
// ErrorHandler class
public static enhanceError($e: \Throwable): \Throwable 
{
  if ($exception instanceof OutOfMemoryError) {
     return $error;
  }

  foreach ($this->getErrorEnhancers() as $errorEnhancer) {
    if ($error = $errorEnhancer->enhance($exception)) {
      return $error;
    }
  }

  return $error;
}
```
and reuse it in both sides.

Reply:
in case of an OutOfMemoryError, enhancing it is likely to break due to needing more memory to compute the enhancements. That's why it is skipped.

Reply:
I was thinking the same things. I'll move the logic

Reply:
Why is framework.handle_all_throwables false by default?